packages/medusa/src/services/product-variant.ts

```diff
@@ -245,86 +254,197 @@ class ProductVariantService extends TransactionBaseService {
     })
   }
 
+  /**
+   * Updates a collection of variant.
+   * @param variantData - a collection of variant and the data to update.
+   * @return resolves to the update result.
+   */
+  async update(
+    variantData: {
+      variant: ProductVariant
+      updateData: UpdateProductVariantInput
+    }[]
+  ): Promise<ProductVariant[]>
+
   /**
    * Updates a variant.
    * Price updates should use dedicated methods.
    * The function will throw, if price updates are attempted.
    * @param variantOrVariantId - variant or id of a variant.
    * @param update - an object with the update values.
-   * @param config - an object with the config values for returning the variant.
    * @return resolves to the update result.
    */
   async update(
     variantOrVariantId: string | Partial<ProductVariant>,
     update: UpdateProductVariantInput
-  ): Promise<ProductVariant> {
+  ): Promise<ProductVariant>
+
+  async update(
+    variantOrVariantId: string | Partial<ProductVariant>,
+    update: UpdateProductVariantInput
+  ): Promise<ProductVariant>
+
+  async update<
+    TInput extends string | Partial<ProductVariant> | UpdateProductVariantData,
+    TResult = TInput extends UpdateProductVariantData
+      ? ProductVariant[]
+      : ProductVariant
+  >(
+    variantOrVariantIdOrData: TInput,
```

Comment:
suggestion: can we split this into 2 or 3 different variables :D the dance we do to identify what this variable is underneath is pretty hard to follow.

Reply:
This is to be compatible, can't do that. Also, the overload allow you to know what the argument can be pretty easily, the implemented method must reflect all overload